Kernel/Process.cpp

```diff
@@ -1077,4 +1077,16 @@ ErrorOr<NonnullRefPtr<Custody>> Process::custody_for_dirfd(int dirfd)
     return *base_description->custody();
 }
 
+StringView Process::name() const
+{
+    return m_name.with([](auto& name) { return name->view(); });
+}
```

Comment:
I feel like we're not going far enough here, as the returned `StringView` is not protected by the `m_name` lock after returning.

Reply:
Oh whoops, yeah!

Reply:
Andreas is right, and the best thing to do here is to just return the SpinlockProtected<NonnullOwnPtr<KString>, ...> like this:
```c++
SpinlockProtected<NonnullOwnPtr<KString> Process::name() const
{
   return m_name;
}
```